### PR TITLE
build: scaffold basic vite app

### PR DESCRIPTION
**Pull request by samsiegart**
refs https://github.com/Agoric/dapp-inter/issues/1

### CODE REVIEW

File: .eslintrc [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+  },
+  "plugins": [
+    "react",
+    "@typescript-eslint",
+    "import",
+    "jsx-a11y",
+    "react-hooks"
+  ],
+  "rules": {
+    "@typescript-eslint/no-explicit-any": "off",
```
Comment (turadg) [inline .eslintrc:29]: +1 to starting with recommended and loosening

File: .gitignore [hunk 2]
```diff
@@ -1,5 +1,25 @@
...
+*.local
+coverage
+
+# Editor directories and files
+.vscode/*
+!.vscode/extensions.json
+.idea
+.DS_Store
+*.suo
+*.ntvs*
```
Comment (turadg) [inline .gitignore:22]: dunno what these are but seems safe

File: .prettierrc [hunk 3]
```diff
@@ -0,0 +1,5 @@
+{
+  "endOfLine": "auto",
+  "singleQuote": true,
+  "arrowParens": "avoid"
```
Comment (turadg) [inline .prettierrc:4]: we need trailingComma to match sdk. it's also the default in Prettier v3 (to come)
```suggestion
  "trailingComma": "all",
  "arrowParens": "avoid"
```

also consider alphabetizing

File: README.md [hunk 4]
```diff
@@ -1,24 +1,20 @@
-# Inter Protocol UI
+# dapp-psm
```
Comment (turadg) [inline README.md:1]: this should still be "Inter Protocol UI"

File: README.md [hunk 5]
```diff
@@ -1,24 +1,20 @@
...
 
-User application for Agoric Inter Protocol--Vaults, BLD Boost, etc.
+UI for Inter Protocol Vaults, BLD Boost, Liquidations, etc.
```
Comment (turadg) [inline README.md:3]: wby lose the dash? it expands the functionality of Inter Protocol

File: README.md [hunk 6]
```diff
@@ -1,24 +1,20 @@
...
-
-## Reusing the code with other parameters
-
-Instead of using the default on-chain settings, you can deploy the
-contract and api by using the scripts in `exampleDeployScripts`. These are
-not intended to be run with the dapp directly as-is, but rather as
-starting points for configuring the deployment in different ways.
+`yarn dev` to start a local HMR server.
+
+By default it it will connect to the prod Wallet UI. To connect to a local one, use
```
Comment (turadg) [inline README.md:9]: please note how this needs a `yarn start` in wallet-app to serve a local network-config

https://github.com/Agoric/dapp-econ-gov#local

File: README.md [hunk 7]
```diff
@@ -1,24 +1,20 @@
...
+http://127.0.0.1:5173/?wallet=local .
+
+## Deployment
+
+(TODO - Add Deployment URL from Cloudflare) serves the latest build of the `beta` branch.
```
Comment (turadg) [inline README.md:14]: this repo doesn't have a `beta` branch.  I think we'll start with `production`, but let's not document that until it's working

File: index.html [hunk 8]
```diff
@@ -0,0 +1,14 @@
+<!DOCTYPE html>
+<html lang="en">
+  <head>
+    <meta charset="UTF-8" />
+    <link rel="icon" type="image/svg+xml" href="/IST.png" />
+    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
+    <meta property="og:image" content="/open-graph.jpg" />
+    <title>Inter Protocol</title>
```
Comment (turadg) [inline index.html:8]: consider,
```suggestion
    <title>Inter Protocol UI</title>
```

File: package.json [hunk 9]
```diff
@@ -1,43 +1,49 @@
...
-    "lint-check": "yarn workspaces run lint-check && (cd ui && yarn lint-check)",
-    "lint": "yarn workspaces run lint-check",
-    "test": "yarn workspaces run test",
-    "build": "yarn workspaces run build"
+    "dev": "vite",
+    "build": "tsc && vite build",
+    "preview": "vite preview",
+    "lint": "eslint src",
+    "format": "prettier --write . && yarn lint --fix",
+    "test": "vitest --passWithNoTests",
```
Comment (turadg) [inline package.json:12]: how about including a simple test so we don't need this option?

File: package.json [hunk 10]
```diff
@@ -1,43 +1,49 @@
...
+    "react": "^18.2.0",
+    "react-dom": "^18.2.0"
   },
-  "prettier": {
-    "trailingComma": "all",
-    "singleQuote": true
+  "devDependencies": {
+    "@endo/eventual-send": "^0.16.0",
+    "@endo/init": "^0.5.43",
+    "@headlessui/react": "^1.6.6",
```
Comment (turadg) [inline package.json:22]: this is as much a full `dependency` as `react` is.  IMO anything that ends up in the build goes in `dependencies` but I'm open to another principle as long as we're consistent

File: tailwind.config.cjs [hunk 11]
```diff
@@ -0,0 +1,43 @@
...
+          '"Segoe UI"',
+          'Roboto',
+          '"Helvetica Neue"',
+          'Arial',
+          '"Noto Sans"',
+          'sans-serif',
+          '"Apple Color Emoji"',
+          '"Segoe UI Emoji"',
+          '"Segoe UI Symbol"',
+          '"Noto Color Emoji"',
```
Comment (turadg) [inline tailwind.config.cjs:31]: seems premature to have all these specified

File: tsconfig.json [hunk 12]
```diff
@@ -0,0 +1,26 @@
+{
+  "compilerOptions": {
+    "target": "ESNext",
+    "useDefineForClassFields": true,
+    "lib": ["DOM", "DOM.Iterable", "ESNext"],
+    "allowJs": false,
+    "skipLibCheck": true,
+    "esModuleInterop": false,
+    "allowSyntheticDefaultImports": true,
+    "strict": true,
```
Comment (turadg) [inline tsconfig.json:10]: 👍

File: vitest.config.ts [hunk 13]
```diff
@@ -0,0 +1,13 @@
+import { mergeConfig } from 'vite';
+import { defineConfig } from 'vitest/config';
+import viteConfig from './vite.config';
+
+export default mergeConfig(
+  viteConfig,
+  defineConfig({
+    test: {
+      setupFiles: ['src/installSesLockdown.ts'],
+      environment: 'happy-dom',
```
Comment (samsiegart) [inline vitest.config.ts:10]: Using jsdom fails on lockdown with `Unexpected intrinsic intrinsics.ArrayBuffer.__proto__`, but luckily the aptly named `happy-dom` works

File: package.json (hunk 10 above)
Comment (samsiegart) [reply package.json:18]: Just tried to sort out the stuff that gets bundled into deps and the rest into dev deps

File: README.md (hunk 6 above)
Comment (samsiegart) [reply README.md]: Done

File: README.md (hunk 7 above)
Comment (samsiegart) [reply README.md]: Removed

File: tailwind.config.cjs (hunk 11 above)
Comment (samsiegart) [reply tailwind.config.cjs]: I think this is okay. These are fallback fonts depending on what the client has available to be used for `font-sans`

File: package.json (hunk 9 above)
Comment (samsiegart) [reply package.json]: Done

File: index.html (hunk 8 above)
Comment (samsiegart) [reply index.html]: Done

File: README.md (hunk 5 above)
Comment (samsiegart) [reply README.md]: Wasn't intentional, fixed. I just copied the readme from dapp-psm and forgot what was here originally

File: README.md (hunk 4 above)
Comment (samsiegart) [reply README.md]: Fixed

File: .prettierrc (hunk 3 above)
Comment (samsiegart) [reply .prettierrc]: Done

File: .eslintrc (hunk 1 above)
Comment (samsiegart) [reply .eslintrc]: Removed

File: test/App.test.tsx [hunk 14]
```diff
@@ -0,0 +1,13 @@
...
+import React from 'react';
+import { expect, it, describe } from 'vitest';
+import { render, screen } from '@testing-library/react';
+import App from '../src/App';
+
+describe('App', () => {
+  it('should render a placeholder message', () => {
+    render(<App></App>);
+
+    expect(screen.getByText(/Hello, Inter Protocol!/i)).toBeDefined();
```
Comment (turadg) [inline test/App.test.tsx:11]: 👍

**Review by turadg — APPROVED** [review]

**Pull request merged**